release/release.sh

```diff
@@ -93,19 +118,36 @@ if ! command -v hub > /dev/null; then
   exit 1
 fi
 
-banner "Starting release for ${PROJECT} (${CLONE_URL})"
-hub clone "${PROJECT}" "${REPO_DIR}"  # May force login to GH at this point
+banner "Starting release for ${PROJECT_ARG} (${CLONE_URL})"
+hub clone "${PROJECT_ARG}" "${REPO_DIR}"  # May force login to GH at this point
 cd "${REPO_DIR}"
 
 # Figures out the most recent tagged version, and computes the next version.
 readonly TAG="$(git describe --tags --abbrev=0 origin/master)"
 readonly CUR_TAG="$(test -n "${TAG}" && echo "${TAG}" || echo "v0.0.0")"
-readonly NEW_RELEASE="$(perl -pe 's/v0.(\d+).0/"v0.${\($1+1)}"/e' <<<"${CUR_TAG}")"
-readonly NEW_TAG="${NEW_RELEASE}.0"
-readonly NEW_BRANCH="${NEW_RELEASE}.x"
+readonly CUR_VERSION="${CUR_TAG#v}"
+
+NEW_VERSION=""
+if [[ -n "${VERSION_ARG}" ]]; then
+  NEW_VERSION="${VERSION_ARG}"
+else
+  # No new version specified; compute the new version number
+  NEW_VERSION="$(perl -pe 's/(\d+).(\d+).(\d+)/"$1.${\($2+1)}.$3"/e' <<<"${CUR_VERSION}")"
+fi
+declare -r NEW_VERSION
+
+# Avoid handling patch releases for now, because we wouldn't need a new branch
+# for those.
+if ! grep -P "\d+\.\d+\.0" <<<"${NEW_VERSION}" > /dev/null; then
```

Comment:
Opt: Consider `case` builtin?
```
case "$1" in
[1-9][0-9]*.[1-9][0-9]*.0)
  :;;
*)
  <error exit>;;
esac
```

Reply:
I think the `if` with grep regex is clearer, IMO, and would prefer to keep it.